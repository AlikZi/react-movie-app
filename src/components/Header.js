import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <div className="header">
    <div className="logo-box container">
    <Link to="/"><img className="logo-image" alt="The Movie DB" src="/images/tmdb.svg" /></Link>
    </div>
  </div>
);

export default Header;