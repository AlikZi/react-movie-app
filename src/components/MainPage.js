import React from 'react';
import '../styles/styles.scss';

const MainPage = () => (
  <div>
    <h1>Movie List</h1>
  </div>
);

export default MainPage;

/*constructor(props) {
    super(props);
    this.componentDidMount = this.componentDidMount.bind(this);
    this.state = {movies: []};
  }

  componentDidMount(){
    fetch(`http://api.themoviedb.org/3/movie/popular?api_key=${API_KEY}&page=1`)
      .then( res => res.json())
      .then( data => this.setState(() => ({movies: data['results']})))
      .catch(err => 
        console.log(err));
    }

    const moviesItems = this.state.movies.map(movie => (
      <div key={movie.id}>
        <p>{movie.title}</p>
        <p>{movie.overview}</p>
        <img class="card-image" alt={movie.title} src={URL_IMAGE + movie.poster_path}/>
      </div>
    ))
*/