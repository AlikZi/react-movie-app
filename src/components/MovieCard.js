import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { URL_IMAGE } from '../api_source';

const MovieCard = ({ movie }) => {
    const {id, title, release_date, vote_average, overview, poster_path } = movie;
    const date = moment(release_date, 'YYYY-MM-DD').format('MMMM D, YYYY');
    const rate = vote_average.toFixed(1);
    return (
    <div className='moviecard'>
    <Link to={`/movie/${id}`}><img className='moviecard-poster' alt={title} src={`${URL_IMAGE}${poster_path}`}/></Link>
        <div className='moviecard-info'>
            <div className='moviecard-info-header'>
                <p className='moviecard-info-header-rate'>{rate}</p>
                <div className='moviecard-info-header-text'>
                    <h3 className='moviecard-info-header-text-title'>{title}</h3>
                    <p className='moviecard-info-header-text-date'>{date}</p>
                </div>
            </div>
            <p className='moviecard-info-overview'>{overview}</p>
            
            <Link style={{ textDecoration: 'none'}} to={`/movie/${id}`}><p className='moviecard-info-link'>More Info</p></Link>
        </div>
    </div>
)};

export default MovieCard;
