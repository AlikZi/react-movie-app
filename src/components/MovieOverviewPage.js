import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import { API_KEY, URL_IMAGE, URL_BACKDROP } from '../api_source';


export default class MovieOverviewPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = { movie: {},
                   videos: [] }
  }
  componentDidMount() {
    // When component mounts scroll to the top of the page
    window.scrollTo(0, 0);
    const id = this.props.match.params.movieId;

    // Fetch details about the movie from TMDB and set the state
    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}`)
      .then( res => res.json())
      .then( (data) => {
        console.log(data);
        this.setState({movie:data})})
      .catch(err => 
      console.log(err));

    // Fetch up to two movie trailers from TMDB and set the state
    fetch(`https://api.themoviedb.org/3/movie/${id}/videos?api_key=${API_KEY}`)
    .then(res => res.json())
    .then( data => {
      // Get only videos with "Trailer" type, save up to two videos in trailers variable
      const trailers = data.results.filter(video => video.type =="Trailer").slice(0, 2);
      this.setState({ videos: trailers})
    })
  }
  render() {
    // Destructure movie for 
    const { 
      title,
      overview,
      release_date,
      backdrop_path,
      poster_path,
      vote_average,
      genres
          }= this.state.movie;
    
    // Format rate to have one decimal
    let rate = vote_average ? vote_average.toFixed(1) : 0.0;

    // Format release date
    const date = moment(release_date, 'YYYY-MM-DD').format('MMMM D, YYYY');

    // Transform the  list of genres into string
    let genres_names = genres ? genres.map(genre => genre.name).join(', ') : [];

    // Create divs that contain iframe with trailer and name below the iframe and save in "videos"
    const videos = this.state.videos.map(video =>{
      return (
      <div key={video.key} className="movie-videos__item">
        <div className="video-container">
            <iframe key={video.key} width="320" height="180" allowFullScreen={true} src={`https://www.youtube.com/embed/${video.key}`} frameBorder="0"/>
        </div>
        <p className="movie-videos__item--name">{video.name}</p>
      </div>
      );
    });

    return (
      <div>
        <Link className="back-link" to={`/`}>
          <img alt="Back Button" className="back-link__button" src="/images/left-arrow.svg"/>
        </Link>
        <div className="backdrop">
          <img className="backdrop__image" alt={title} src={`${URL_BACKDROP}${backdrop_path}`}/>
        </div>
        <div className='overview'>
          <div className="container">
            <div className="poster-title-box">
              <img className="poster-title-box__poster" alt={title} src={`${URL_IMAGE}${poster_path}`}/>
              <div className="poster-title-box__info">
                <h2 className="poster-title-box__info--title">{title}</h2>
                <p className='poster-title-box__info--date'>{date}</p>
                <p className='poster-title-box__info--genres'>{genres_names}</p>
              </div>
              <p className="poster-title-box__rate">{rate}</p>
            </div>
            
            <div className="overview-text">
                <h3>Overview</h3>
                <p>{overview}</p>
            </div>
            <h3 className="videos-header">Trailers</h3>
            <div className="movie-videos">
                {videos}
            </div>
          </div>
        </div>
      </div>
    );
  }
};
