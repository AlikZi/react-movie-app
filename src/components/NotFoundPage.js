import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div className="container">
    <h1>404 - NOT FOUND</h1>
    <h2><Link to="/">Go home</Link></h2>
  </div>
);

export default NotFoundPage;