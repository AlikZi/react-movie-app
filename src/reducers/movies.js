const moviesReducerDefaultState = [];

export default (state = moviesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_MOVIES' :
            return [
                ...state,
                ...action.movies
            ];
        default:
            return state;
    }
}