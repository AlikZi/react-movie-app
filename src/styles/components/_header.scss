.header {
    background-color: $secondary-color;
    width: 100%;
    height: 12rem;
    //position: fixed;
}

.logo-box {
    height: 12rem;
    display: grid;
    justify-items: start;
    align-content: center;
    
}

.logo-image{
    height: 8rem;
}