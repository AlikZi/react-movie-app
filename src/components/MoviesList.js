import React from 'react';
import { connect } from 'react-redux';
import MovieCard from './MovieCard';

export const MoviesList = (props) => (
    <div>
        {
            // Check if movies are passed down, if not show the message
            props.movies.length === 0 ? (
                <h1>Sorry, we are facing technical issues at the moment. Please, try again later.</h1>
            ) : (
                <div className="movie-list">
                    {props.movies.map(movie => (
                        <MovieCard key={movie.id} movie={movie}/>))}
                </div>
            )
        }

    </div>
)

// Get movies from redux store
const mapStateToProps = (state) => {
    return {
        movies: state
    }
};

// Connect component to redux store
export default connect(mapStateToProps)(MoviesList);
