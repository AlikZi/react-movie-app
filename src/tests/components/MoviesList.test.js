import React from 'react';
import { shallow } from 'enzyme';
import { MoviesList } from '../../components/MoviesList';
import movies from '../fixtures/movies';

test('Should render MoviesList with given movies', () => {
    const wrapper = shallow(<MoviesList movies={movies} />)
    expect(wrapper).toMatchSnapshot();
});

test('Should render MoviesList without movies', () => {
    const wrapper = shallow(<MoviesList movies={[]} />);
    expect(wrapper).toMatchSnapshot();
})