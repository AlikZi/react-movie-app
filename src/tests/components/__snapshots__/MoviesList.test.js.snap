// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MoviesList with given movies 1`] = `
<div>
  <div
    className="movie-list"
  >
    <MovieCard
      key="1"
      movie={
        Object {
          "id": 1,
          "overview": "First lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum",
          "poster_path": "/1.jpg",
          "release_date": 1970,
          "title": "First Movie",
          "vote_average": 5.5,
        }
      }
    />
    <MovieCard
      key="2"
      movie={
        Object {
          "id": 2,
          "overview": "Second lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum",
          "poster_path": "/2.jpg",
          "release_date": 1985,
          "title": "Second Movie",
          "vote_aveage": 7.9,
        }
      }
    />
    <MovieCard
      key="3"
      movie={
        Object {
          "id": 3,
          "overview": "Third lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum",
          "poster_path": "/3.jpg",
          "release_date": 1982,
          "title": "Third Movie",
          "vote_aveage": 7.5,
        }
      }
    />
  </div>
</div>
`;

exports[`Should render MoviesList without movies 1`] = `
<div>
  <h1>
    Sorry, we are facing technical issues at the moment. Please, try again later.
  </h1>
</div>
`;
